zerver/views/registration.py

```diff
@@ -528,6 +563,15 @@ def registration_helper(
         realm_user_default = RealmUserDefault.objects.get(realm=realm)
         default_email_address_visibility = realm_user_default.email_address_visibility
 
+    selected_realm_type_name = None
+    if realm_creation:
+        assert prereg_realm is not None
+        selected_realm_type_name = [
+            org_type["name"]
+            for org_type in Realm.ORG_TYPES.values()
+            if org_type["id"] == prereg_realm.org_type
+        ][0]
```

Comment:
Should we move this to a helper? This function is already huge and nearly impossible to mentally parse, so seems nice to encapsulate additional fiddly logic elsewhere :neutral_face:

Reply:
Yeah that's a great idea.

Reply:
Added a helper. We can also directly add the below field to context object instead of adding a new function -
```python
"selected_realm_type_name": None if prereg_realm is None else get_selected_realm_type_name(prereg_realm),
```
Can do this change, if we find it easily readable.